modules/hal_nordic/nrfx/Kconfig

```diff
@@ -286,6 +286,7 @@ config NRFX_RNG
 
 config NRFX_RRAMC
 	bool "RRAMC driver"
+	depends on !TRUSTED_EXECUTION_NONSECURE
 	depends on $(dt_has_compat,$(DT_COMPAT_NORDIC_RRAM_CONTROLLER))
```

Comment:
This is not a question of nrfx driver support but of hardware design, so I think this should be rather addressed in dts, similarly to how it is done for `uicr` or `wdt30`: https://github.com/zephyrproject-rtos/zephyr/blob/85f36aae51e5c6dfc38de2427f60e9c72897342d/dts/common/nordic/nrf54l_05_10_15.dtsi#L647-L650
Then the existing `depends on` will work as expected.

Reply:
> Because [cpuapp_rram](https://github.com/zephyrproject-rtos/zephyr/blob/85f36aae51e5c6dfc38de2427f60e9c72897342d/dts/common/nordic/nrf54l_05_10_15.dtsi#L707) is part of rram_controller

And this seems to be another thing that should be corrected. The `*_rram` nodes should be children of the `soc` node, not the `rram-controller` one; just like `uicr`.

Reply:
@anangl Just to clarify: Are you talking about reusing the define set here:

https://github.com/nrfconnect/sdk-zephyr/blob/f4cb3034fc6db139d2917b2f0fff2946ac034e90/boards/nordic/nrf54l15dk/nrf54l15dk_nrf54l15_cpuapp_ns.dts#L9

Or are you suggesting adding another #define for the purpose?

Reply:
> @anangl Just to clarify: Are you talking about reusing the define set here:
> 
> https://github.com/nrfconnect/sdk-zephyr/blob/f4cb3034fc6db139d2917b2f0fff2946ac034e90/boards/nordic/nrf54l15dk/nrf54l15dk_nrf54l15_cpuapp_ns.dts#L9

Yes.